---
title: "06_knitting_ATAC_analyses"
author: "JR"
date: "2024-11-30"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
source("/scratch/Shares/rinnclass/MASTER_CLASS/STUDENTS/reanna/MASTER_CLASS/util/00_our_first_function_lesson.R")
library(GenomicRanges)
library(IRanges)
library(dplyr)
library(tidyr)
library(tibble)
library(readr)
library(ggplot2)
library(purrr)
library(magrittr)
library(pheatmap)
library(textshape)
library(Rcpp)
library(DESeq2)
library(stringr)
```

#### Objective: To determine if chromatin accessibility changes due to dox exposure.
We have performed a time course series of experiements measuring chromating accessibility
(ATACseq peaks)upon exposure to dox. These are 0, 30, 60, 90, 120, 150 minutes.
Each time point has a replicate. The fastq sequencing files were processed by the
NF_CORE ATACseq pipeline v- 2.12 (https://nf-co.re/atacseq/2.1.2/).

We will be using the MACS2 output files from the NF_CORE Pipeline For example:
-broad.peak
-consensus.broad.peak
-consensus.feature.counts

#### Approach:

##### (a) Loading in ATACseq peak files with custom function import_peaks (list of GRanges output)

```{r loading in peak files to list of GRanges}

# establishing peak path to the dir with MACS2 output peak files from NF_CORE ATACseq pipeline
#peak_path <- "<your_atac_pipeline_output_dir>/bwa/merged_library/macs2/broad_peak"
peak_path <- "/scratch/Shares/rinnclass/MASTER_CLASS/lessons/08_ATACseq_pipeline/00_run_pipeline/00_pipeline_run/pipeline_run/bwa/merged_library/macs2/broad_peak"

# creating a file list also needed for import_peaks function to get sample name associated with file
fl <- list.files(peak_path, full.names = TRUE, pattern = ".broadPeak")

# running import_peaks
my_peaks <- import_peaks(consensus_file_path = peak_path)

print("here are the number of peaks for each sample")
print(num__peaks <- sapply(my_peaks, length) %>% as.data.frame)

```
##### Result - there are roughly 70-100K peaks in a given sample

##### (b) finding number of peaks common in all samples using find_common_peaks custom funciton
Here I am asking how many peaks are called in all the time points measured. These are an interesting
set of genomic locations as these chromatin accessibility sites persist across all samples. 
```{r finding common peaks in all samples, dependson="previous_chunk"}
# run find_common_peaks function 
common_peaks <-  suppressWarnings(find_common_peaks(my_peaks))

```
Here are the number of peaks that overlap in all samples: `r length(common_peaks)`

![example of peak called by find_common_peaks in all samples.](results/common_peak_example.jpg)
##### The figure shows that our common peak call looks good. Peaks appear in all conditions.
##### Result 37,152 peaks are common in all samples - visual inspection in IGV looks good (see below)


##### (c) Peaks that are unique to dox and non-dox conditions
Using find_common_overlaps peaks that are specific to dox or non-dox
will be identified.

Non-dox samples
```{r non-dox atac peaks, dependson="previous_chunk"}

# making DESEQ sample sheet with peak files
# common peaks in non-dox (0 time point)
non_dox_samples <- my_peaks[c("KO_control_0", "WT_control_0")]
non_dox_common_peaks <- suppressWarnings(find_common_peaks(non_dox_samples))
print(c("This is how many peaks are common in non-dox:", length(non_dox_common_peaks)))
```

Dox samples
```{r dox samples, dependson="previous_chunk"}
# common peaks in dox time points (!not time 0)
dox_samples <- names(my_peaks)[!grepl("_0$", names(my_peaks))]
dox_peaks <- my_peaks[dox_samples]
dox_common_peaks <- suppressWarnings(find_common_peaks(dox_peaks))
print(c("This is how many peaks are common in dox",length(dox_common_peaks)))
```

Overlap of dox and non-dox (to get unique to each condition)
```{r dox vs non-dox atac peaks, dependson="previous_chunk"}
# Now overlap between dox and non-dox common peaks
dox_compare_list <- list(non_dox = non_dox_common_peaks, dox = dox_common_peaks)
dox_non_dox_ov <- suppressWarnings(find_common_peaks(dox_compare_list))
print(c("This is how many peaks are common in both non- and dox treatments", length(dox_non_dox_ov)))

# extracting peaks unique to each condition (dox non-dox)
# Peaks unique to non_dox
unique_to_non_dox <-suppressWarnings(find_my_peaks(dox_non_dox_ov, non_dox_common_peaks))
print(c("This is how many peaks are unique to non-dox condition",length(unique_to_non_dox)))

# Peaks unique to dox
unique_to_dox <- suppressWarnings(find_my_peaks(dox_non_dox_ov, dox_common_peaks))

print(c("This is how many peaks are unique to dox condition", length(unique_to_dox)))
```

Number of peaks that unique to non-dox condition: `r length(unique_to_non_dox)`
![Example figure of a non-dox unique peak (not too convincing).](results/common_peak_example.jpg)

Note in this figure that the common peak in all samples does not look much different than
the one unique to dox condition. Suspicious - and no statistics can be used - overlaps may
not be the best approach.

Number of peaks that unique to dox condition: `r length(unique_to_dox)`
![example of peak unique to dox examples.](results/dox_specifc_example.jpg)

#### Summary dox non-dox overlaps 

##### Result: almost all peaks common in dox were also present in non-dox
##### Result: 37435 dox common peaks of the 38984 non-dox peaks overlapped (96%)
##### Result: 16312 peaks found in non-dox that were not found in dox common
##### Result: 1549 unique peaks in dox and not present in non-dox
##### Result: Overall bias to more non-dox specific peaks (could it be due to sample num?)

##### (d) Creating mouse gene, lincrna, mRNA annotation GRange objects
Here I am going to create GRange objects of genome annotations from 
Gencode mv25. Specifically we will create gene annotation GRanges and 
their corresponding promoter region. These objects will be used for overlaps with ATAC peaks
```{r creating genome annotation GRanges, dependson="previous_chunk"}

# Loading gencode genome annotation as GRanges 
gencode_gr_mouse <- rtracklayer::import("/scratch/Shares/rinnclass/MASTER_CLASS/GENOMES/M25/gencode.vM25.annotation.gtf")

# all gene promoters
gencode_genes_mouse <- gencode_gr_mouse[gencode_gr_mouse$type == "gene"] 
gene_promoters <- promoters(gencode_genes_mouse, upstream = 2000, downstream = 2000)

# mrna_genes
mrna_genes <- gencode_genes_mouse[gencode_genes_mouse$gene_type %in% "protein_coding"]
mrna_promoters <- promoters(mrna_genes, upstream = 2000, downstream = 2000)

# lincrna_genes
lincrna_genes <- gencode_genes_mouse[gencode_genes_mouse$gene_type %in% "lincRNA"]
lincrna_gene_promoters <- promoters(lincrna_genes, upstream = 2000, downstream = 2000)

# This is more reliable than looking at all peaks because these regions in the genome are more stable
```

##### (e) Compare overlaps of dox and non-dox peaks with gene annotaitons
Now we will overlap our dox and non-dox unique peaks with genome annotations (gene promoters)
First we will find number of overlaps with gene promoters and then genes that had changed in RNAseq
```{r dox and non-dox unique peak overalp gene promoters, dependson="previous_chunk"}

# gr_list of promoters and peaks unique to non_dox condition
gr_list_gene_promoter_non_dox_ov <- list( gene_promoters = gene_promoters, non_dox_peaks = unique_to_non_dox)
non_dox_gene_promoter_ov <- suppressWarnings(find_common_peaks(gr_list_gene_promoter_non_dox_ov))

print("This is how many non-dox_unique peaks overlapped gene promoters")
length(non_dox_gene_promoter_ov)
# 2178 genes had an overlap with an ATAC peak at the zero time point but not in dox treated samples

# peaks unique to dox condition overlapped with gene promoters
gr_list_gene_promoter_dox_ov <- list( gene_promoters = gene_promoters, dox_peaks = unique_to_dox)
dox_gene_promoter_ov <- suppressWarnings(find_common_peaks(gr_list_gene_promoter_dox_ov))

print(c("This is how many dox_unique peaks overlapped gene promoters", length(dox_gene_promoter_ov)))
# 239 genes are unique to only dox

# Now find same gene_id in RNAseq resuts
# Loading RNAseq results from 06_Differential_expression_analyses
load("/scratch/Shares/rinnclass/MASTER_CLASS/lessons/06_Differential_expression_analyses/results/counts_results_DESEQ/DESEQ_results.rdata")

# filter significant genes from RNAseq "filtered_res_df" to non-dox unique promoter overlaps
# res df and filtered res df are the results we need to use (filtered by padj)
sig_rnaseq_atac_non_dox <- non_dox_gene_promoter_ov[non_dox_gene_promoter_ov$gene_id %in% filtered_res_df$gene_id]

print(c("this is how many genes overlap between RNAseq and ATACseq non-dox peaks", length(sig_rnaseq_atac_non_dox)))
# 47 genes have no peak at zero time point but peaks appeared after dox treatment

print(c("Here are the gene names that overlap", sig_rnaseq_atac_non_dox$gene_name))

# filter significant genes from RNAseq "filtered_res_df" to non-dox unique promoter overlaps
sig_rnaseq_atac_dox <- non_dox_gene_promoter_ov[dox_gene_promoter_ov$gene_id %in% filtered_res_df$gene_id]

print(c("this is how many genes overlap between RNAseq and ATACseq non-dox peaks", length(sig_rnaseq_atac_dox)))
# 37 genes had a peak at zero time point but peaks disappeared after dox treatment
```
Number of genes overlap between RNAseq and ATACseq non-dox peaks: `r length(sig_rnaseq_atac_dox)

![example of peak overlapping gene changing by RNAseq not to convincing again ....](results/RNAseq_atac_overlap.jpg)
# should use IGV to include pictures in the final project as well
Also note that the gene that is called unique to non-dox doesnt' look different than common peak to all

##### Summary:
##### Result: 2178 non-dox peaks overlap gene promoters out of 16,312 (13%)
##### Result: 239 dox peaks pverlap gene promoters out of 1,793 (13% same as non-dox) 
##### Result: 47 and 37 genes had overlap with genes that also changed by RNAseq
##### Overall: shows our overlap approaches worked but not great for finding differential peaks

### WARNING: Visual inspection of peaks does not find this approach convincing
Importing bigWig files and peak calls showed very little convincing data that
peaks were changing in dox and non-dox conditions. Does show good overlaps with 
gene annotations so approach is working well.

##### (f) use DESEQ2 to find out if peaks are changing in dox and non-dox conditions
The overlap analysis did not contain statistical analysis and is a logical approach
that ended up not being that compeling based on raw data. So now we will use a 
statistical approach DESEQ2 to compare peak read counts across samples of dox and non-dox conditions.
To do so we will use consensus peaks (any peak called in any condition) and feature counts of each 
consensus peak to be used as input into DESEQ2 for differential expression of peak counts

```{r create input files and run DESEQ2 on feature counts of consensus peaks from NF_CORE, dependson="previous_chunk"}

# consensus peaks
broad_consensus_peaks <- read.table("/scratch/Shares/rinnclass/MASTER_CLASS/lessons/08_ATACseq_pipeline/00_run_pipeline/00_pipeline_run/pipeline_run/bwa/merged_library/macs2/broad_peak/consensus/consensus_peaks.mLb.clN.annotatePeaks.txt",
                             sep = "\t", header = TRUE)

# consensus peak counts
broad_consensus_counts <- read.table("/scratch/Shares/rinnclass/MASTER_CLASS/lessons/08_ATACseq_pipeline/00_run_pipeline/00_pipeline_run/pipeline_run/bwa/merged_library/macs2/broad_peak/consensus/consensus_peaks.mLb.clN.featureCounts.txt",
                             sep = "\t", header = TRUE)

# could get counts matrix from these files by cleaning them up via putting stuff in meta

# creating sample sheet "atac_samples" from file names (col names of consensus):
rownames(broad_consensus_counts) <- broad_consensus_counts$Geneid

broad_consensus_counts <- broad_consensus_counts %>%
  dplyr::select(-c(Geneid, Chr, Start, End, Strand, Length))
colnames(broad_consensus_counts) <- gsub("_REP1",
                                         "_R1",
                                         gsub(
                                           "\\.mLb\\.clN\\.sorted\\.bam",
                                           "",
                                           colnames(broad_consensus_counts)
                                         ))
count_columns <- colnames(broad_consensus_counts)

atac_samples <- data.frame(
  sample = count_columns,
  condition = ifelse(
    grepl("_0_", count_columns),
    "non-dox",
    ifelse(grepl("non-dox", count_columns), "non-dox", "dox")
  ),
  timepoint_minutes = as.numeric(sub(".*_(\\d+)_R1.*", "\\1", count_columns))
)

# atac_samples is a sample sheet for DESEQ now!

# Factor condition for DESEQ2 !!
atac_samples <- atac_samples %>%
  mutate(condition = factor(condition, levels = c("non-dox", "dox")))

# matrix for Deseq2
atac_dds_condition <- suppressWarnings(DESeqDataSetFromMatrix(countData = broad_consensus_counts, 
                                   colData = atac_samples, 
                                   design = ~ condition))
# Run DESeq2 condition model
atac_dds_condition <- suppressWarnings(DESeq(atac_dds_condition))

```

##### (g) extracting and analyzing DESEQ2 results
Creating a dataframe with pvlaues and lfc etc 
```{r extracting DESEQ2 results, dependson="previous_chunk"}

# Extract DESeq2 results
# print the results from DESEQ which is the intercept file in the comparisonand make it a data frame
atac_lfc_condition <- results(atac_dds_condition) %>%
  as.data.frame() %>%
  rownames_to_column("interval_id")

# Merge with broad_consensus_peaks info

colnames(broad_consensus_peaks)[1] <- "interval_id"

atac_lfc_condition <- merge(atac_lfc_condition, 
                  broad_consensus_peaks %>%
                    dplyr::select(interval_id, Gene.Name, Nearest.PromoterID, 
                                  Distance.to.TSS, Chr, Start, End),
                  by = "interval_id")

```
![example of DESESEQ2 SIG ATAC PEAK top statistical](results/top_statistical_hit_deseq.jpg)
Result: Not very convincing

![example of DESESEQ2 SIG ATAC PEAK high base-mean](results/example_high_basemean_sig.jpg)
Result: Not very convincing

##### (h) analyzing results of DESEQ on ATACseq peak counts
Now we will see how many peaks had a padj < 0.05 and 
other features of our analyses!

```{r analysis of DESEQ2 on ATAC peak counts, dependson="previous_chunk"}

hist(atac_lfc_condition$padj)
# almost all padj are near 1 !! none at 0.05

hist(atac_lfc_condition$pvalue)
# Ok some pvals are sig but worrisome that padj no

hist(atac_lfc_condition$log2FoldChange)
# Ok some good changes ranging from log2 -5 to 5

hist(atac_lfc_condition$baseMean)
# all very low - perhaps expected for ATACseq vs RNAseq

```
# Result : no peaks are significant by padj criterion
# Warning: visual inspection does not show these peaks to look compelling 
### Result: Chromatin accessibility does not appear to reliably or robustustly change upon dox exposure.
## keymatching is important in data science

#### Conclusion (1) Interesting set of common chromatin accessibility sites
There are ~30,000 peaks that are consistent across all samples. This is interesting as these are 
robust and reliable sites of chromatin accessibility in mouse ESCs. Meaning epigenetic markers of the
mESC cell state. It maybe an interesting future approach to investigate how many of these sites changed 
in RNA expression. This may indicate if regulatory events were more biased to open sites? Also we could 
simply compare expression levels of genes (even if they don't change in time) compared to these common sites 
versus non common sites.

#### Conclusion (2) Peaks do not sigificantly change due to dox
In neither overlap approach nor DESEQ2 were any peaks significant. 
Visual inspection of all tests failed to find compelling peaks. 
More importantly, there were no signifcant statics that could be used to define changing peaks.

#### Conclusion (3) RNA gene expression changes due to dox but not chromatin accessibility!
This is quite interesting that we observe 908 genes changing in RNA expression. However,
we don't find significant ATAC peak changes. Thus dox must work in a non-epigenetic mechanism. Thus,
dox could be influencing post transcriptional or transcriiptional changes but not epigenetic.

##########################################
Exercise: Knit document into web document!
##########################################

# just click the "knit" button (bawl of string above .rmd pane)

Voila !! I hope your proud you just condensed this level into 300 lines of code!
Now your ready to start using all you have learned on even more data (same samples but human stem cells!)

# Congratulations and now level up to final lv 10 !

# final project ideas: run atac seq for human data to confirm it doesn't change, could see if conserved genes are both up and downregulated, take peaks 