---
title: "02_pipeline_outputs"
author: "JR"
date: "2024-11-07"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# GOAL to explore the output files from the ATACseq pipeline !
Congratulations, if you are here it means you have completed your 
ATACseq pipeline run !

Outputs found at: /scratch/Shares/rinnclass/MASTER_CLASS/lessons/08_ATACseq_pipeline/00_run_pipeline/00_pipeline_run/pipeline_run

Now let's look through the results step by step !

# (1) QC analysis
There is a multiqc .html report - transfer to your comp and open - it's here:
08_ATACseq/00_run_pipeline/00_pipeline_run/pipeline_run/multiqc/broad_peak

bigwig files for IGV are in bwa folder along with macs2 (every possible peak file). consensus folder is in macs2 folder.
/scratch/Shares/rinnclass/MASTER_CLASS/lessons/08_ATACseq_pipeline/00_run_pipeline/00_pipeline_run/pipeline_run/bwa/merged_library/macs2/broad_peak/consensus

# Look through and answer these questions:
- How many reads per sample?

- Did the X or Y chromosome get more reads mapped?

- what was roughly the average insert size (size of DNA from ATACseq)?

- what can you conclude from the "Read Distribution Profile after Annotation" ?

* MACS2 section has a bunch of cool results:
- How many peaks were called (estimate)?
- What percentage of reads were inside peaks?
    - hint: FRiP score
    - what do you make of this?
    
- What type of annotation (HOMER) did most peaks land on?

- Which category do you think has the smallest genome space but most peaks considering ...?

- The PCA plot tells you how similar the samples are to each other
    - There is an important trend that maybe of CONCERN?
    
- Interestingly the heat map doesn't look as bad as the PCA plot.
    - red means smaller distance or more similar.
- If you had to say there are two trends, how many samples in each category?

# (2) RESULTS - Raw Data (BigWig) !
In your pipeline output directory there is a folder called BWA.
This has the key results we need.

Follow these steps and answer the questions along the way !

- Transfer "bigwig" files to your computer to look at with IGV.
- Find 2 good examples of peaks 
    - (hint your peak file is in MACS2 > BroadPeak > Consensus folder - )
    - good examples would be peak in all samples or one changing by time


# (3) RESULTS -MACS2 (Model-based Analysis for ChIP-Seq))

Go to the consensus folder and transfer "intersect.plot.pdf" only PDF in folder
Try and understand the plot.

What sample has the most unique number of peaks for just that sample.
Which two have the most peaks in common
How many peaks are in all samples?

Explore even more in IGV or in the result output and enjoy !

########### Thought ExeRcise ###########  

# (1) What are consensus peaks and how are they made by MACS2?

# (2) What research questions could we ask with the consensus peaks?

# (3) Look up Broad Peak / Narrow Peak / gapped peak 

# Hint: Use ChatGPT :) just make sure to use MACS2 in prompt !

Next we will move on to explore where the peaks are in the genome
and how they compare across time points. 




